packages/core/content-type-builder/admin/src/utils/parseDateValue.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema } from '@strapi/types';
+
+export const parseDateValue = <TAttribute extends Schema.Attribute.AnyAttribute>(
+  value: Schema.Attribute.Value<TAttribute>
```

Comment:
To be honest, I think the input type can be simplified to `unknown` here since `Schema.Attribute.Value<TAttribute>` will resolve to `any` most of the time anyway (impossible intersection between too many different types).

The function signature would then look like
```ts
declare function parseDateValue(value: unknown): Date | undefined;
```

As for https://github.com/strapi/strapi/pull/21140#discussion_r1743862547, I would argue manually defining the return type in that kind of utils ensure we don't deviate from what is expected to be returned in the future, but that's a matter of preferences and it's not that important, whatever you think is best here.

Reply:
I'll update the input type then :)
And for the return type, if it were up to me, I'd prefer to declare it  manually just for better readability :D 
Do we have a norm for typescript usage?

Reply:
We're in the ongoing process of writing them :upside_down_face: